etna/packages/etna-js/dispatchers/work-dispatcher.jsx

```diff
@@ -1,17 +1,27 @@
-import { createWorker } from '../upload/workers/index';
+import {createWorker, terminateWorker} from '../upload/workers/index';
 import { WORK, WORK_FAILED } from '../upload/actions/upload_actions';
 
 // Export for testing
 export const WORKERS = {
   upload: (dispatch) => createWorker( dispatch, require.resolve('../upload/workers/uploader'))
 }
 
+// For tests to disable all active workers and prevent test polution of async operations.
+const clearAllWorkersQueue = [];
+
 // a middleware that dispatches commands to worker threads
 const workDispatcher = () => {
   let active_workers = { };
 
+
+  function clearAllWorkers() {
+    Object.values(active_workers).forEach(terminateWorker)
+  }
+
+  clearAllWorkersQueue.push(clearAllWorkers);
+
   return store => next => action => {
-    let { type, work_type, ...args } = action;
+    let { type, work_type, command } = action;
```

Comment:
This made debugging a bit easier, since the command re the worker is received in self containment, rather than having to unpack it away from the other properties.